jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactoryInternals.java

```diff
@@ -52,6 +52,9 @@
         PathResourceFactory pathResourceFactory = new PathResourceFactory();
         RESOURCE_FACTORIES.put("file", pathResourceFactory);
         RESOURCE_FACTORIES.put("jrt", pathResourceFactory);
+
+        if (NativeImagePathResourceFactory.ENABLE_GRAALVM_RESOURCE_SCHEME)
+            RESOURCE_FACTORIES.put("resource", new NativeImagePathResourceFactory());
```

Comment:
I'm still cautious about this.  If this was not a VM feature, then I would definitely not like this.

The alternative is to have a module that adds in this mapping "resource" to the resource factory, but then embedded code would still need to do a registration explicitly.

Reply:
We could easily add other Path/FileSystem implementation schemes automatically here.

I would probably access something in Jetty that is known, and use it's protocol/scheme to register other schemes at this point in time.

Eg:

``` java
ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
        URL url = classLoader.getResource("org/eclipse/jetty/version/build.properties");
        if (url != null)
        {
            RESOURCE_FACTORIES.put(url.getProtocol(), pathResourceFactory);
        }
```

Reply:
Isn't a `newFileSystem` call still needed somewhere?    Doesn't need to be in a `ResourceFactory` but is still needed right?

Reply:
The `newFileSystem` call is now made from within the `NativeImagePathResourceFactory` constructor.

Reply:
@gregw AFAICT, there is only 1 root FileSystem for the `resource:` scheme, representing all of the content in the entire native-image.
So having the `NativeImagePathResourceFactory` control the mount makes sense here.
But a more general approach (for any other FileSystem types) is definitely something to seek out.